app/controllers/application_records_controller.rb

```diff
@@ -22,19 +22,27 @@ def create
   def csv_export
     start_date = parse_american_date(params.require :start_date)
     end_date = parse_american_date(params.require :end_date)
-    params.require :department_ids
-    @records = ApplicationRecord.in_department(params[:department_ids])
-               .between(start_date, end_date)
+    if params[:department_ids]
```

Comment:
Well, it seems like the general job of the logic you need is to come up with a list of department IDs to pass to your various methods based on the value of the `department_ids` parameter. Then you can just pass that list to the method in question. If you have further questions I'll be on Slack.